Migrate from Circle CI to GitHub Actions

colincasey opened the pull request:
[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
Merge branch 'main' into migrate_to_gh_actions

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey commented on pom.xml:
```diff
@@ -159,6 +159,9 @@
                             <goal>jar</goal>
                         </goals>
+                        <configuration>
+                            <doclint>all,-syntax</doclint>
```
@Malax this following gets the build to pass on Java 11 runs but is really just covering up the underlying issue.  

from the [debug output i got the javadoc plugin to spit out](https://github.com/forcedotcom/sf-fx-sdk-java/actions/runs/3388506081/jobs/5630569985) (see the `cat` commands after the failure) there was no reason i could find that would explain why it would want to treat the `@ParametersAreNonnullByDefault` annotation in `package-info.java` as a javadoc tag.  i was not able to reproduce the error locally either.  

what's your take on this?

Malax replied:
Interesting. I think we should get to the bottom of this - if it happens during release too we will be unable to release. Adding `-syntax`, we might mask other errors we'd care about. I can take a closer look tomorrow.

Malax commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,24 @@
+      fail-fast: false
+      matrix:
+        version: [11, 8]
+        os: [windows-latest, ubuntu-latest, macos-latest]
```
Nit: Not sure if we need to test builds on all OSes here, given the nature of this repo.

colincasey replied:
write once, run everywhere?  i don't trust it 😉

Malax commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,24 @@
+    strategy:
+      fail-fast: false
+      matrix:
+        version: [11, 8]
```
Should we add `17` here too? (latest LTS) - wasn't in the original file so nothing we need to fix here, just an observation.

Malax commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,24 @@
+          java-version: ${{ matrix.version }}
+          distribution: zulu
+          cache: maven
+      - run: ./mvnw --no-transfer-progress clean package
```
`--no-transfer-progress` should probably just be `--batch` which has the same effect.

Malax reviewed (changes requested):
"Requesting changes" for the outstanding doclint issue

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)

colincasey committed:
feat: migrate from Circle CI to GitHub Actions

[W-11838222](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000017dCK4YAM/view)
